statsmodels/tsa/statespace/sarimax.py

```diff
@@ -0,0 +1,1941 @@
+        coefficients on the exogenous regressors are allowed to vary over time.
+        Default is False.
+    mle_regression : boolean, optional
+        Whether or not to use estimate the regression coefficients as part of
+        maximum likelihood estimation or through the Kalman filter (i.e.
+        recursive least squares). If `time_varying_regression` is True, this
+        must be set to False. Default is True.
+    simple_differencing : boolean, optional
+        Whether or not to use conditional maximum likelihood estimation.
+        If True, differencing is performed prior to estimation, which discards
+        the first :math:`SD + d` initial rows but reuslts in a smaller
+        state-space formulation. If False, the full SARIMAX model is put in
+        state-space form so that all datapoints can be used in estimation.
+        Default is False.
+    enforce_stationarity : boolean, optional
+        Whether or not to transform the AR parameters to enforce stationarity
+        in the autoregressive component of the model. Default is True.
+    enforce_invertibility : boolean, optional
+        Whether or not to transform the MA parameters to enforce invertibility
+        in the moving average component of the model. Default is True.
+    hamilton_representation : boolean, optional
+        Whether or not to use the Hamilton representation of an ARMA process
+        (if True) or the Harvey representation (if False). Default is False.
+
+    Attributes
+    ----------
+    measurement_error : boolean
+        Whether or not to assume the observations were measured with error.
+    state_error : boolean
+        Whether or not the transition equation has an error component.
+    mle_regression : boolean
+        Whether or not to use estimate the regression coefficients as part of
+        maximum likelihood estimation.
+    state_regression : boolean
+        Whether or not to use estimate the regression coefficients via the
+        Kalman filter, by extending the state space.
+    time_varying_regression : boolean
+        Whether or not coefficients on the exogenous regressors are allowed to
+        vary over time. Not yet implemented.
+    simple_differencing : boolean
+        Whether or not to use conditional maximum likelihood estimation.
+    enforce_stationarity : boolean
+        Whether or not to transform the AR parameters to enforce stationarity
+        in the autoregressive component of the model.
+    enforce_invertibility : boolean
+        Whether or not to transform the MA parameters to enforce invertibility
+        in the moving average component of the model.
+    hamilton_representation : boolean
+        Whether or not to use the Hamilton representation of an ARMA process.
+    trend : str{'n','c','t','ct'} or iterable
+        Parameter controlling the deterministic trend polynomial :math:`A(t)`.
+        See the class parameter documentation for more information.
```

Comment:
explicit keywords are fine since we use it almost everywhere.
I thought about the internal storage, instead of having all of them attached as attributes, attach a dict as attribute
`self.model_specification = dict('measurement_error' = measurement_error, ....)`
or even an OrderedDict or a Bunch. It would be easier to hand around and see the specification in one place.

maybe you leave the explicit attributes in the model and add only the model-specification dict to the results instance.

related: In other models we added a similar list of keyword that are arguments in `__init__` so there is a simple generic way to figure out which keywords to add when recreating a model automatically, e.g. we want to internally get a changed new model for auxiliary results.